src/wasm.rs

```diff
@@ -74,44 +76,85 @@ impl Future for InnerFuture {
 }
 
 mod fetch {
+    use futures_util::io::AsyncReadExt;
     use js_sys::{Array, ArrayBuffer, Reflect, Uint8Array};
     use wasm_bindgen::JsCast;
     use wasm_bindgen_futures::JsFuture;
-    use web_sys::window;
-    use web_sys::RequestInit;
+    use web_sys::{window, RequestInit};
 
     use std::io;
     use std::iter::{IntoIterator, Iterator};
+    use std::pin::Pin;
 
     /// Create a new fetch request.
-    pub(crate) fn new(method: impl AsRef<str>, url: impl AsRef<str>) -> Request {
-        Request::new(method, url)
-    }
 
     /// An HTTP Fetch Request.
     pub(crate) struct Request {
-        init: RequestInit,
-        url: String,
+        request: web_sys::Request,
+        _body_buf: Pin<Vec<u8>>,
```

Comment:
This name should not have a leading `_` since we're using it.
```suggestion
        body_buf: Pin<Vec<u8>>,
```

Reply:
This strictly isn't being used in Rust, it's only being held to ensure the buffer isn't deallocated while the request is completed in Javascript. However, I also don't like the leading `_`, so I changed the name and added a `#[allow(dead_code)]` to suppress the warning.